### PR TITLE
v1.12 Backports 2023-09-04

### PR DESCRIPTION
**Pull request by jibi**
* [ ] #27656 (@pchaigno)
 * [x] #27572 (@bimmlerd)
 * [x] #27814 (@haiyuewa)
 * [x] #27359 (@brb)
 * [x] #27706 (@tklauser)

PRs skipped due to conflicts:

 * #27831 (@tklauser)
 * #27798 (@ti-mo)
   * dropped while `lb4_extract_tuple()` gets backported to 1.12

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 27656 27572 27814 27359 27706; do contrib/backporting/set-labels.py $pr done 1.12; done
```
or with
```
make add-labels BRANCH=v1.12 ISSUES=27656,27572,27814,27359,27706
```

### CODE REVIEW

File: Documentation/gettingstarted/servicemesh/grpc.rst [hunk 1]
```diff
@@ -20,14 +20,14 @@ For this demo we will use `GCP's microservices demo app <https://github.com/Goog
...
-    $ kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/master/release/kubernetes-manifests.yaml
+    $ kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/main/release/kubernetes-manifests.yaml
 
 Since gRPC is binary-encoded, you also need the proto definitions for the gRPC
 services in order to make gRPC requests. Download this for the demo app:
 
 .. code-block:: shell-session
 
-    $ curl -o demo.proto https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/master/pb/demo.proto
+    $ curl -o demo.proto https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/main/pb/demo.proto
```
Comment (haiyuewa) [inline Documentation/gettingstarted/servicemesh/grpc.rst]: "main/pb" --> "main/protos"

File: Documentation/gettingstarted/servicemesh/tls-termination.rst [hunk 2]
```diff
@@ -143,7 +143,7 @@ Make HTTPS Requests
         .. code-block:: shell-session
 
             # Download demo.proto file if you have not done before
-            $ curl -o demo.proto https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/master/pb/demo.proto
+            $ curl -o demo.proto https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/main/pb/demo.proto
```
Comment (haiyuewa) [inline Documentation/gettingstarted/servicemesh/tls-termination.rst]: "main/pb" --> "main/protos"

**Review by haiyuewa — CHANGES_REQUESTED** [review]

File: Documentation/gettingstarted/servicemesh/grpc.rst [hunk 3]
```diff
@@ -20,14 +20,14 @@ For this demo we will use `GCP's microservices demo app <https://github.com/Goog
...
-    $ kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/master/release/kubernetes-manifests.yaml
+    $ kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/main/release/kubernetes-manifests.yaml
 
 Since gRPC is binary-encoded, you also need the proto definitions for the gRPC
 services in order to make gRPC requests. Download this for the demo app:
 
 .. code-block:: shell-session
 
-    $ curl -o demo.proto https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/master/pb/demo.proto
+    $ curl -o demo.proto https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/main/proto/demo.proto
```
Comment (haiyuewa) [inline Documentation/gettingstarted/servicemesh/grpc.rst]: miss the "s", should be "main/protos"  : - )

File: Documentation/gettingstarted/servicemesh/tls-termination.rst [hunk 4]
```diff
@@ -143,7 +143,7 @@ Make HTTPS Requests
         .. code-block:: shell-session
 
             # Download demo.proto file if you have not done before
-            $ curl -o demo.proto https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/master/pb/demo.proto
+            $ curl -o demo.proto https://raw.githubusercontent.com/GoogleCloudPlatform/microservices-demo/main/proto/demo.proto
```
Comment (haiyuewa) [inline Documentation/gettingstarted/servicemesh/tls-termination.rst]: miss the "s", should be "main/protos"  : - )

**Review by haiyuewa — CHANGES_REQUESTED** [review]

**Review by tklauser — APPROVED** [review]
My change looks good, thanks.

> * [Fix propagation of namespace labels to CEP labels #27831](https://github.com/cilium/cilium/pull/27831) (@tklauser)

Not applicable for v1.12, I've removed the `need-backport/1.12` label.

**Review by bimmlerd — APPROVED** [review]
#27572 LGTM, thanks!

**Review by qmonnet — CHANGES_REQUESTED** [review]
- https://github.com/cilium/cilium/pull/27798

`lb4_extract_tuple()` does not exist on v1.12, making the commit broken. I'm currently looking at the 1.12 backport for this PR.

**Comment by jibi** [comment]
> * [bpf: lxc: support Pod->Service->Pod hairpinning with endpoint routes #27798](https://github.com/cilium/cilium/pull/27798)
> 
> 
> `lb4_extract_tuple()` does not exist on v1.12, making the commit broken. I'm currently looking at the 1.12 backport for this PR.

ack, I'll drop the backport for now then :+1:

**Comment by jibi** [comment]
/test-backport-1.12

**Comment by jibi** [comment]
/test-backport-1.12

**Review by haiyuewa — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Comment by jibi** [comment]
/test-1.23-4.19

**Comment by jibi** [comment]
/test-1.17-4.9

**Comment by jibi** [comment]
/test-1.21-4.9

**Comment by jibi** [comment]
/test-1.17-4.9

**Review by brb — APPROVED** [review]
Thanks!

**Review by joestringer — APPROVED** [review]
Ack for @cilium/github-sec

**Review by youngnick — APPROVED** [review]

**Pull request merged**